x-pack/plugins/data_visualizer/public/application/index_data_visualizer/hooks/use_overall_stats.ts

```diff
@@ -243,13 +243,16 @@ export function useOverallStats<TParams extends OverallStatsSearchStrategyParams
             nonAggregatableFields
           );
 
+          const documentCountStats = processDocumentCountStats(
+            documentCountStatsResp?.rawResponse,
+            searchStrategyParams
+          );
+
           setOverallStats({
-            documentCountStats: processDocumentCountStats(
-              documentCountStatsResp?.rawResponse,
-              searchStrategyParams
-            ),
+            documentCountStats,
             ...nonAggregatableOverallStats,
             ...aggregatableOverallStats,
+            totalCount: documentCountStats?.totalCount ?? aggregatableOverallStats.totalCount,
```

Comment:
If there are no aggregatable fields in the index, the doc count will still be `0`.

Reply:
You're right and the previous fix wouldn't work when there's no aggregatable fields. Previously, in fact, if there's no time field defined, we wouldn't show the total count or the event rate chart at all. I've fixed it to use track_total_hits here https://github.com/elastic/kibana/pull/134083/commits/cd38416daf50f1761e9ec40cb503b601013e40c9 inside the event rate chart query so that this number will show up accurately, even when there's no time field defined or no aggregatable field